parquet-hadoop/src/main/java/org/apache/parquet/hadoop/MemoryManager.java

```diff
@@ -155,4 +169,37 @@ long getTotalMemoryPool() {
   float getMemoryPoolRatio() {
     return memoryPoolRatio;
   }
+
+  /**
+   * Register callback and deduplicate it if any.
+   * @param callBackName the name of callback. It should be identical.
+   * @param callBack the callback passed in from upper layer, such as Hive.
+   */
+  public void registerScaleCallBack(String callBackName, Runnable callBack) {
+    Preconditions.checkNotNull(callBackName, "callBackName");
+    Preconditions.checkNotNull(callBack, "callBack");
+
+    if (callBacks.containsKey(callBackName)) {
+      throw new IllegalArgumentException("The callBackName " + callBackName +
+          " is duplicated and has been registered already.");
+    } else {
+      callBacks.put(callBackName, callBack);
+    }
+  }
+
+  /**
+   * Get the registered callbacks.
+   * @return
+   */
+  Map<String, Runnable> getScaleCallBacks() {
+    return callBacks;
```

Comment:
Could this return an UnmodifiableMap instead of the one used by the memory manager itself?